contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -73,153 +76,211 @@ contract LidoOracle is ILidoOracle, AragonApp {
-        if (1 == members.length) {
-            QUORUM_POSITION.setStorageUint256(1);
-        }
-
         emit MemberAdded(_member);
-        _assertInvariants();
     }
 
     /**
      * @notice Remove `_member` from the oracle member committee
      * @param _member Address of a member to remove
      */
     function removeOracleMember(address _member) external auth(MANAGE_MEMBERS) {
-        require(members.length > getQuorum(), "QUORUM_WONT_BE_MADE");
-
         uint256 index = _getMemberId(_member);
         require(index != MEMBER_NOT_FOUND, "MEMBER_NOT_FOUND");
-
-        uint256 lastReportedEpochId = (
-            LAST_REPORTED_EPOCH_ID_POSITION.getStorageUint256()
-        );
-
-        MIN_REPORTABLE_EPOCH_ID_POSITION.setStorageUint256(lastReportedEpochId);
         uint256 last = members.length.sub(1);
-
-        uint256 bitMask = gatheredEpochData[lastReportedEpochId].reportsBitMask;
-        if (index != last) {
-            members[index] = members[last];
-            bitMask = bitMask.setBit(index, bitMask.getBit(last));
-        }
-        bitMask = bitMask.setBit(last, false);
-        gatheredEpochData[lastReportedEpochId].reportsBitMask = bitMask;
-
+        if (index != last) members[index] = members[last];
         members.length--;
-
         emit MemberRemoved(_member);
-        _assertInvariants();
+
+        // Nullify the data for the last epoch. Remained oracles will report it again
+        REPORTS_BITMASK_POSITION.setStorageUint256(0);
+        delete gatheredReportKinds;
     }
 
     /**
      * @notice Set the number of oracle members required to form a data point to `_quorum`
      */
     function setQuorum(uint256 _quorum) external auth(MANAGE_QUORUM) {
-        require(members.length >= _quorum && 0 != _quorum, "QUORUM_WONT_BE_MADE");
-
+        require(0 != _quorum, "QUORUM_WONT_BE_MADE");
         QUORUM_POSITION.setStorageUint256(_quorum);
         emit QuorumChanged(_quorum);
+        _tryPush(REPORTABLE_EPOCH_ID_POSITION.getStorageUint256());
+    }
 
-        uint256 minReportableEpochId = (
-            MIN_REPORTABLE_EPOCH_ID_POSITION.getStorageUint256()
-        );
-        uint256 lastReportedEpochId = (
-            LAST_REPORTED_EPOCH_ID_POSITION.getStorageUint256()
-        );
-
-        assert(lastReportedEpochId <= getCurrentEpochId());
-
-        if (lastReportedEpochId > minReportableEpochId) {
-            minReportableEpochId = lastReportedEpochId;
-            _tryPush(lastReportedEpochId);
-        }
+    /**
+     * @notice Returns the callback contract address to be called upon quorum
+     */
+    function getQuorumCallback() public view returns(address) {
+        return address(QUORUM_CALLBACK_POSITION.getStorageUint256());
+    }
 
-        _assertInvariants();
+    /**
+     * @notice Set the callback contract address to be called upon quorum
+     */
+    function setQuorumCallback(address _addr) external auth(SET_QUORUM_CALLBACK) {
```

Comment:
Better to check that address is not 0x0

Reply:
0x0 is "unset". A valid value. Should be commented though.